src/nvim/globals.h

```diff
@@ -1228,6 +1228,8 @@ EXTERN FILE *time_fd INIT(= NULL);  /* where to write startup timing */
 EXTERN int ignored;
 EXTERN char *ignoredp;
 
+EXTERN int  disable_char_avail_for_testing INIT(= 0);
```

Comment:
We've avoid this garbage up to now by using lua unit or funtional tests instead. And it's not needed here either, because we have a real API. Vim needs it because Vim tests Vim with Vim, doesn't have asynchronous input, and doesn't have screen tests capable of testing intermediate states.